portals/devportal/source/src/app/components/Apis/Details/GraphQLConsole/AddEditAdditionalHeaders.jsx

```diff
@@ -0,0 +1,194 @@
+                error = fieldValue === '' ? fieldName + ' is Empty' : false;
+                break;
+            case 'value':
+                error = fieldValue === '' ? fieldName + ' is Empty' : false;
+                break;
+            default:
+                break;
+        }
+        return error;
+    };
+    const formHasErrors = (validatingActive = false) => {
+        if (hasErrors('name', name, validatingActive)
+        || hasErrors('value', value, validatingActive)) {
+            return true;
+        } else {
+            return false;
+        }
+    };
+    const formSaveCallback = () => {
+        setValidating(true);
+        if (!formHasErrors(true)) {
+            return ((setOpen) => {
+                callBack({ name, value }, item);
+                setOpen(false);
+            });
+        }
+        return false;
+    };
+    return (
+        <FormDialogBase
+            title={item === undefined
+                ? `${intl.formatMessage({
+                    id: 'GraphQL.Devportal.Tryout.Additional.header.add.new',
+                    defaultMessage: 'Add New ',
+                })}`
+                : `${intl.formatMessage({
+                    id: 'GraphQL.Devportal.Tryout.Additional.header.edit',
+                    defaultMessage: 'Edit ',
+                })}`}
+            saveButtonText={intl.formatMessage({
+                id: 'GraphQL.Devportal.Tryout.Additional.header.dialog.btn.save',
+                defaultMessage: 'Save',
+            })}
+            triggerButtonText={item ? null : intl.formatMessage({
+                id: 'GraphQL.Devportal.Tryout.Additional.header.dialog.trigger.add',
+                defaultMessage: 'Add',
+            })}
+            icon={item ? <CreateIcon /> : null}
+            triggerButtonProps={{
+                color: 'default',
+                variant: 'contained',
+                size: 'small',
+            }}
+            formSaveCallback={formSaveCallback}
+        >
+            <DialogContentText>
+                <Typography variant='caption'>
+                    <FormattedMessage
+                        id='GraphQL.Devportal.Tryout.Addtional.headers.help'
+                        defaultMessage='This configuration is used to add additional Headers.'
+                    />
+                </Typography>
+            </DialogContentText>
+            <TextField
+                margin='dense'
+                name='name'
+                value={item === undefined ? null : name}
+                onChange={onChange}
+                label={(
+                    <span>
+                        <FormattedMessage
+                            id='GraphQL.Devportal.Tryout.Additional.header.form.name'
+                            defaultMessage='Header Name'
+                        />
+                        <span className={classes.error}>*</span>
+                    </span>
+                )}
+                fullWidth
+                multiline
```

Comment:
don't think we need multiple lines for the header name

Reply:
+1